src/common/messagesHelper.mo

```diff
@@ -0,0 +1,41 @@
+import HashMap "mo:base/HashMap";
+import Text "mo:base/Text";
+import CommonTypes "./types";
+
+module {
+  public class Messages() {
+    private var entries : [(Text, Text)] = [
+      ("ProviderNotInAllowList", "registerProvider - Provider not in allow list with provider ID: "),
+      ("SubmitHtmlContent", "submitHtmlContent sourceId: "),
+      ("GetTasks", "getTasks - provider called with provider ID: "),
+      ("GetTasksFinished", "getTasks - FINISHED - provider called with provider ID: "),
+      ("Vote", "vote - User ID: "),
+      ("PohCallbackForModclub", "pohCallbackForModclub - status: "),
+      ("GetCanisterLog", "getCanisterLog - request from caller: "),
+      ("VotingCompleted", "Voting completed for packageId: "),
+      ("VotingCompletedApproved", "Voting completed. Final decision: approved for packageId: "),
+      ("VotingCompletedRejected", "Voting completed. Final decision: rejected for packageId: "),
+      ("UserVoting", "User Voting "),
+      ("UserIDVoting", "UserID: RS Before Vote POH: Full rewards"),
+      ("PreUpgrade", "MODCLUB PREUPGRADE at time: "),
+      ("PostUpgrade", "MODCLUB POSTUPGRADE at time: "),
+      ("HttpRequest", "http_request - called for url "),
+      ("HttpRequestUpdate", "http_request_update - called for url "),
+      ("ImportAirdropMetadataCall", "MODCLUB Instance has importAirdropMetadata call: "),
+      ("ImportAirdropMetadataProfiles", "MODCLUB Instance has importAirdropMetadata call: ")
+    ];
+
+    private var messages = HashMap.fromIter<Text, Text>(entries.vals(), 16, Text.equal, Text.hash);
+
+    public func logMessage(logger: CommonTypes.ModclubLogger, key : Text, additionalInfo : Text) {
```

Comment:
I believe `logger` can be set in construction step and used later for any `logMessage` call. Similar approach you can find with authGuard class, it accepts env once and uses it all the time.
also additionalInfo can be provided as an array(list of TExt values), and main text can contain a placeholders like `{<index_in_array>}` so it will looks like `{1}`.
Such approach provides us with necessary flexibility in text formatting( that was one of major goals for the refactoring).

Reply:
> {<index_in_array>} so it will looks like {1}

There's no support for proper string interpolation in Motoko, don't wanna complicate it any further

Reply:
Not sure for now, I'll have a separate issue create for it